### ISSUE TITLE
Build fails on Raspberry Pi 4, looking for rexxsaa

### ISSUE THREAD

**Issue by jeff-snyder** [wrljet/hercules-helper#17]
Hi Bill,

I am working to build Hercules on my Pi 4.  Thanks for your script! I started out using Professor Freland's process until I ran into problems with it, your script makes this all much simpler.  However, I ran into a couple minor issues and a problem I'd like to report.

Issue 1: This is just a minor documentation issue in your readme file.  Where you show the "hyperion-buildall.sh" commands, you prefix them with a tilde, "~".  Wouldn't it be better to use ".."?  In my case, my hercules-helper folder is not in my home directory (/home/jeff), but a level lower (/home/jeff/Hercules/hercules-helper).  When I run the commands with ".." from the herctest directory the scripts appear to run fine (well, up to the error listed below).

The other issue I ran into was in they hyperion-buildall.sh script.  I had a network issue that caused the the git clones to fail and the script did not detect the failure, instead prompting me to move forward into the "bldlvlck" step.
I have included the log for you to see what I mean and maybe find a way to capture the errors, if possible.

[log-1.txt](https://github.com/wrljet/hercules-helper/files/7190734/log-1.txt)

The problem I encountered is that when I try to build hercules, using either "--verbose --prompts" or "--auto", the script fails with the following errors trying to find the rexxsaa.h file.  FYI, I did a "find / -name rexxsaa.h" in an attempt to manually find the file with no luck either.  REXX appears to work with no errors, though I don't use it, so my simplistic test didn't do much (say 'hello').

**"Check REXX and compiler files" step:**

Step: Check REXX and compiler files:  Hit return to continue

REXX presence    : /usr/bin/rexx
Checking for Regina-REXX...  
Found REXX       : REXX-Regina_3.6 5.00 31 Dec 2011
Regina version   : 3.6
cc_status = 1
**[[ error: ]] rexxsaa.h is not found in cc search path**
++ <stdin>:1:10: fatal error: rexxsaa.h: No such file or directory
#include "rexxsaa.h"
Checking for ooRexx... nope
Found REXX, but not ooRexx

**"Configure" step:**

ERROR: Regina Rexx support requested but rexxsaa.h is not available 
configure: error: Please correct the above error(s) and try again 
**[[ error: ]] configure failed!**

At this point the script stops running.

I have included log-2 (buildall with verbose and prompts) and log-3 (buildall with auto) for your review.  FYI, I have saved all 3 herctest directories that relate to the submitted log files, in case we need to dig into something a little deeper.

Note: I don't need REXX, so if there is an option to turn it off that would be fine for me, but for other users of your script we should probably figure out why this is failing.

[log-2.txt](https://github.com/wrljet/hercules-helper/files/7190735/log-2.txt)
[log-3.txt](https://github.com/wrljet/hercules-helper/files/7190736/log-3.txt)

My system is: 

jeff@Pi4:~ $ uname -a
Linux Pi4.Snyder.ORG 5.10.60-v7l+ #1449 SMP Wed Aug 25 15:00:44 BST 2021 armv7l GNU/Linux

jeff@Pi4:~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

Please let me know if I can provide any additional information.  Thanks!

**Comment by jeff-snyder** [comment]
I found the following in "https://sdl-hercules-390.github.io/html/rexx.html":

_If the headers rexx.h and oorexxapi.h are found, then support for OORexx will be provided. If the header rexxsaa.h is found, then support for Regina Rexx will be provided. If all three headers are found, then support for both OORexx and Regina will be provided and you can then specify at runtime which one you want Hercules to use. (See the 'rexx' "start" option further below.)_

So the script should not be trying to install Regina REXX support, since the file rexxsaa.h is not present, right?

Also, I found the "--no-rexx" option on the hyperion-buildall script (jeez, RTFM much? :) ) and tried it, but I still get the failure.  Again, the script is trying to install it when I specifically said not to.

Is it possible there's a logic error in the script, around deciding whether to install Regina REXX?

**Comment by wrljet** [comment]
I will look into it.

Bill

**Comment by jeff-snyder** [comment]
Hi Bill,

Thanks for looking at this.

Always glad to help!  I'm good at finding errors... :-)

I reread the READMEs and the note you added about Regina REXX looks good.

I installed the Regina development package and a buildall with --verbose and --prompt worked.  It found REXX and rexxsaa.h and included support for Regina REXX in the build.  I ran for an hour or so using that hercules and did not find any errors.

After rereading the documentation and log files, I realize I misunderstood the meaning of the "--no-rexx" option switch.  I was thinking the option meant to not include support for REXX in Hercules when it appears to mean "Don't build Regina itself".  When I look at the log for a run with "--no-rexx", it says "skip : Build Regina REXX".    Since REXX is found, the script also logs a message that says, "Regina REXX is present.  Skipping build from source" on all the runs, so that seems to confirm my new understanding.

Jeff
